collector.go

```diff
@@ -163,24 +164,38 @@ func (c *Collector) Collect(ch chan<- prometheus.Metric) {
 					log.Printf("Error collecting job status metrics: %v\n", err)
 					continue
 				}
-				for _, j := range r.JobSummaryList {
-					// Use DescribeJobs for each JobId to get detailed info, including the JobDefinition
-                    describeRes, err := c.client.DescribeJobsWithContext(ctx, &batch.DescribeJobsInput{
-                        Jobs: []*string{j.JobId},
-                    })
-                    if err != nil {
-                        log.Printf("Error fetching job details for JobId %s: %v\n", *j.JobId, err)
-                        continue
-                    }
-
-					// Get JobDefinition from detailed result
-					definition := "undefined"
-					if len(describeRes.Jobs) > 0 && describeRes.Jobs[0].JobDefinition != nil {
-						// Get simplified definition out of the ARN
-						definition = getJobDefinitionSubstring(*describeRes.Jobs[0].JobDefinition)
+
+				// Collect Job IDs for job description
+				var jobIDs []*string
+				for _, job := range r.JobSummaryList {
+					jobIDs = append(jobIDs, job.JobId)
+				}
+
+				// We need to process this in chunks because of AWS hard limit on DescribeJobsWithContext
+				var batchJobIDs []*string
+				for i := 0; i < len(jobIDs); i += maxDescribeSize {
+					end := i + maxDescribeSize
```

Comment:
We have go1.23, we could use this instead of 4 lines:
```go
end := min(i + maxDescribeSize, len(jobsIDs))
```
But I'll let you decide if that is easier to read or not